packages/SwingSet/src/kernel/liveSlots.js

```diff
@@ -532,25 +536,31 @@ function build(
   // vats which use D are in: acorn-eventual-send, cosmic-swingset
   // (bootstrap, bridge, vat-http), swingset
 
-  // here we finally invoke the vat code, and get back the root object
-  const rootObject = buildRootObject(
-    harden({ D, exitVat, exitVatWithFailure, ...vatPowers }),
-    harden(vatParameters),
-  );
-  mustPassByPresence(rootObject);
+  const vatGlobals = harden({}); // for later use: #1846
+
+  function setBuildRootObject(buildRootObject) {
+    assert(!didRoot);
+    didRoot = true;
```

Comment:
This is entirely aesthetic, but it bugs me a little to see the `didRoot` flag set immediately before calling `buildRootObject` instead of immediately after.

Reply:
Hm, if `buildRootObject()` were to throw an exception, would we want to let them retry? I think the answer is "no"?

Reply:
That's a very good point and totally correct, but it still bothers me to have a flag assert that you did a thing that you haven't actually done yet.  Even though it's only a lie for one line of execution.  I did say it was 100% aesthetic after all.  You could rename the flag (`irrevocablyCommitedToAttemptToBuildRoot` or some such) to satisfy this concern, but I've already gotten you seriously into fitting the noise here so totally don't sweat it.

Reply:
More seriously wondering now if `setBuildRootObject` is a good name for this function, since it doesn't set something called `buildRootObject` but is instead given it to immediately invoke.  Isn't what's getting set is the root object itself, not the builder function, so maybe it's really more like `buildAndSetRootObject`?

Reply:
Yeah, I wasn't entirely happy with that name. The root object is definitely built while calling this function, but the vat can't build that object by itself (`buildRootObject()` takes arguments, which must come from liveslots). So `setRootObject(rootObject)` wouldn't work, even though it'd be more accurate. And the arguments which will be passed to `buildRootObject()` are details that I wouldn't want the vat manager/worker setup stuff to have to observe, so it wouldn't be ideal to have `makeLiveSlots` e.g. return `{ vatPowers, vatParameters }` and then ask the three worker-factories to use them in a call to `buildRootObject()` itself.

I guess I can't think of a better name right now, but if you think of one, I'd be happy to change it.